### PR TITLE
Fixed incorrect timeout message when test host crashes/exits

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
Fixed incorrect timeout message when test host crashes/exits

Related issue
#2022

### CODE REVIEW

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/ProxyExecutionManagerTests.cs [hunk 1]
```diff
@@ -338,6 +340,22 @@ public void SetupChannelShouldThrowExceptionIfClientConnectionTimeout()
...
+            this.mockTestHostManager.Setup(tmh => tmh.LaunchTestHostAsync(It.IsAny<TestProcessStartInfo>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(true)).Callback(() => { this.mockTestHostManager.Raise(t => t.HostExited += null, new HostProviderEventArgs("I crashed!")); });
+
+            try
+            {
+                this.testExecutionManager.SetupChannel(new List<string> { "source.dll" });
+            }
+            catch (TestPlatformException tpe)
+            {
+                Assert.AreEqual(string.Format(CrossPlatEngineResources.Resources.TestHostExitedWithError, "I crashed!"), tpe.Message);
+            }
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/ProxyExecutionManagerTests.cs]: Use Assert.ThrowException, it returns message, assert for message separately if you want.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs [hunk 2]
```diff
@@ -161,6 +161,7 @@ public virtual bool SetupChannel(IEnumerable<string> sources)
                     !this.RequestSender.WaitForRequestHandlerConnection(connTimeout * 1000, this.CancellationTokenSource.Token))
                 {
                     this.CancellationTokenSource.Token.ThrowTestPlatformExceptionIfCancellationRequested();
+                    this.ThrowOnTestHostExited(this.testHostExited.IsSet);
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs]: Add comments for each case

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/ProxyExecutionManagerTests.cs (hunk 1 above)
Comment (ShreyasRmsft) [reply test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/ProxyExecutionManagerTests.cs]: Didn't know about the return value. Thanks. will make this change

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs (hunk 2 above)
Comment (ShreyasRmsft) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs]: okay

**Review by singhsarab — APPROVED** [review]
:shipit:

**Pull request merged**